src/lib/CustomRulePathManager.ts

```diff
@@ -94,6 +94,54 @@ export class CustomRulePathManager extends AsyncCreatable {
 		return classpathEntries;
 	}
 
+	public async getMatchingPaths(language: string, paths: string[]): Promise<string[]> {
+		await this.initialize();
+
+		this.logger.trace(`Returning paths for language ${language} that match patterns [${paths}]`);
+
+		// Expand the patterns into actual paths. E.g., expand directories into the rule objects they contain, etc.
+		const expandedPaths = await this.expandPaths(paths);
+
+		// Now that we've got the possible paths, we need to see which ones are actually present.
+		return expandedPaths.filter((p) => {
+			// Determine the engine associated with this path.
+			const e = this.determineEngineForPath(p);
+			// If there's nothing mapped for that engine, or the engine has nothing for this language, we can drop this
+			// path.
+			if (!this.pathsByLanguageByEngine.has(e) || !this.pathsByLanguageByEngine.get(e).has(language)) {
```

Comment:
Thinking out loud. Highly possible that the user adds a language in lowercase and tries to delete passing in the same language in mixed case. As of today, we do this only in the Java layer. We should rather be unifying the case before storing in CustomPath.json.

Reply:
You're proposing always casting the language argument to lowercase? That makes sense to me.